src/js/views/right_panel_view.js

```diff
@@ -1326,18 +1387,29 @@
             const isVerticalFlipped = $button.hasClass('active');
 
             this.models.forEach(function(m) {
-                m.save('vertical_flip', isVerticalFlipped);
+                var shapes = m.get('shapes');
+                if(shapes){
+                    let newShapes = shapes.map(sh => {
+                        if (sh.type === "Text") {
+                            return {...sh, vFlip: isVerticalFlipped ? -1 : 1};
+                         }
+                        return sh;
+                    });
+                    m.save({'vertical_flip': isVerticalFlipped, 'shapes': newShapes});
+                }else{
+                    m.save('vertical_flip', isVerticalFlipped);
+                }
```

Comment:
This block can be reverted since the `panel_view.js` now updates the ShapeManager directly on render_shapes (as for isHorizontalFlipped below)